lib/logger/lib/logger/translator.ex

```diff
@@ -307,17 +328,26 @@ defmodule Logger.Translator do
       is_integer(args) ->
         format_undef(mod, fun, args, undef, stacktrace)
       true ->
-        Exception.format_exit(reason)
+        format_stop_banner(undef)
     end
   end
 
-  defp format_otp_exit(reason) do
-    Exception.format_exit(reason)
+  defp format_stop_banner(reason, stacktrace) do
+    if Exception.exception?(reason) do
+        Exception.format_banner(:error, reason, stacktrace)
+    else
+      case Exception.normalize(:error, reason, stacktrace) do
+        %ErlangError{} ->
+          format_stop_banner(reason)
```

Comment:
My main thoughts are:

* I would like not to "guess" as much as possible. Whenever I do some error formatting code in Elixir I have this voice in the back of my mind worrying that error handling/formatting is not a clean abstraction. But then I remember if it wasn't like this it would be like Erlangs...

My other thoughts are:

* I think the decision here should be followed in Exception module too
* I don't think we can only special case undefined function errors. What about function clause? What about case clause? etc.
* I am concerned this is a step away from seamless interop, i.e. how easy is it for an Elixir "newbie" to just use an Erlang library without knowing Erlang?
* If we were to take this path then wouldn't it require an Elixir implementation of GenServer and a wrapper for Kernel and Process spawn/hibernate functions to normalize errors before exiting? So that Elixir code "always" produces Elixir exceptions.
* We will always have to do the "try format stacktrace" dance.